server/service/labels.go

```diff
@@ -71,12 +72,26 @@ func (svc *Service) NewLabel(ctx context.Context, p fleet.LabelPayload) (*fleet.
 	}
 	label.Name = p.Name
 
-	if p.Query != "" && (len(p.Hosts) > 0 || len(p.HostIDs) > 0) {
```

Comment:
Shouldn't we fail the request early if the user attempts to set two of `Query`, `Hosts` or `Criteria`?

Reply:
yes, done!